WalletWasabi/Blockchain/Keys/KeyManager.cs

```diff
@@ -682,9 +685,23 @@ public void AssertNetworkOrClearBlockState(Network expectedNetwork)
 	private static HdPubKey CreateHdPubKey((KeyPath KeyPath, ExtPubKey ExtPubKey) x) =>
 		new(x.ExtPubKey.PubKey, x.KeyPath, Analysis.Clustering.LabelsArray.Empty, KeyState.Clean);
 
-	internal void SetExcludedCoinsFromCoinJoin(IEnumerable<OutPoint> excludedOutpoints)
+	public void UpdateExcludedCoinsFromCoinJoin(IEnumerable<OutPoint> addedCoinsToExclusion, IEnumerable<OutPoint> removedCoinsFromExclusion)
+	{
+		var addedCoinsToExclusionArray = addedCoinsToExclusion as OutPoint[] ?? addedCoinsToExclusion.ToArray();
+		var removedCoinsFromExclusionArray = removedCoinsFromExclusion as OutPoint[] ?? removedCoinsFromExclusion.ToArray();
+
+		ExcludedCoinsFromCoinJoin.AddRange(addedCoinsToExclusionArray);
+		ExcludedCoinsFromCoinJoin.RemoveAll(removedCoinsFromExclusionArray.Contains);
+
+		BypassAutomaticExclusionFromCoinjoin.AddRange(removedCoinsFromExclusionArray);
+		BypassAutomaticExclusionFromCoinjoin.RemoveAll(addedCoinsToExclusionArray.Contains);
+
+		ToFile();
+	}
```

Comment:
The logic seems easy here but is not. I've thought about it for quite some time and this current solution is clever: One list allows you to specifically exclude coins, the other to specifically include coins. Those lists are only updated manually. That way the user can retain full control on its UTXOs even if he has an automatic exclusion threshold.